rc/filetype/org.kak

```diff
@@ -0,0 +1,78 @@
+        haml haskell html ini java javascript json julia kak kickstart latex
+        lisp emacs-lisp scheme lua makefile org moon objc perl pug python ragel ruby rust
+        sass scala scss sh swift toml tupfile typescript yaml sql
+    "
+    for lang in ${languages}; do
+        printf "%s\n" "add-highlighter shared/org/${lang} region '#\+(?i)BEGIN_SRC(?I)\h+${lang}\b' '#\+(?i)END_SRC' regions"
+        printf "%s\n" "add-highlighter shared/org/${lang}/ default-region fill meta"
+        case ${lang} in
+            kak)        ref="kakrc"   ;;
+            emacs-lisp) ref="lisp"    ;;
+            *)          ref="${lang}" ;;
+        esac
+        printf "%s\n" "add-highlighter shared/org/${lang}/inner region \A#\+(?i)BEGIN_SRC(?I)[^\n]*\K '(?=#\+(?i)END_SRC)' ref ${ref}"
+    done
+}
+
+# Comment
+add-highlighter shared/org/comment region (^|\h)\K#[^+] $ fill comment
+
+# No-language SRC block
+add-highlighter shared/org/src_block region ^(\h*)(?i)#\+BEGIN_SRC(\h+|\n) ^(\h*)(?i)#\+END_SRC\h*$ fill meta
+
+# Example block
+add-highlighter shared/org/example_block region ^(\h*)(?i)#\+BEGIN_EXAMPLE(\h+|\n) ^(\h*)(?i)#\+END_EXAMPLE\h*$ fill mono
+
+# Small example block
+add-highlighter shared/org/inline/text/example regex ^(\h*)[:]\h+[^\n]* 0:mono
+
+# Unordered list items start with `-', `+', or `*'
+add-highlighter shared/org/inline/text/unordered-lists regex ^(?:\h*)([-+])\h+ 1:bullet
+# But `*' list must be indented with at least single space, if not it is treated as a heading
+add-highlighter shared/org/inline/text/star-list regex ^(?:\h+)([*])\h+ 1:bullet
+
+# Ordered list items start with a numeral followed by either a period or a right parenthesis, such as `1.' or `1)'
+add-highlighter shared/org/inline/text/ordered-lists regex ^(?:\h*)(\d+[.)])\h+ 1:bullet
+
+# Headings
+add-highlighter shared/org/inline/text/heading    regex "^[*]+\h+[^\n]+"                          0:header
+add-highlighter shared/org/inline/text/italic     regex "\s([/][^\s/].*?[^\s/]*?[/])\W" 1:default,default+i
+add-highlighter shared/org/inline/text/verbatim   regex "\s([=][^\s=].*?[^\s=]*?[=])\W" 1:meta
+add-highlighter shared/org/inline/text/code       regex "\s([~][^\s~].*?[^\s~]*?[~])\W" 1:mono
+add-highlighter shared/org/inline/text/math       regex "\s([$][^\s].*?[^\s]*?[$])\W"   1:mono
+add-highlighter shared/org/inline/text/underlined regex "\s([_][^\s_].*?[^\s_]*?[_])\W" 1:default,default+u
+add-highlighter shared/org/inline/text/bold       regex "\s([*][^\s*].*?[^\s*]*?[*])\W" 1:default,default+b
+add-highlighter shared/org/inline/text/link       regex "\[[^\n]+\]\]"                            0:link
+add-highlighter shared/org/inline/text/option     regex "(?i)#\+[a-z]\w*\b"                       0:meta
+
+# Strikethrough highlighter is disabled because Kakoune doesn't support strikethrough attribute
+# add-highlighter shared/org/inline/text/strikethrough regex "\W*\+[^\s][^\n+]+[^\s]\+\W" 0:default,default+s
```

Comment:
You should still define the regex so it can be highlighted with a colour, even if Kakoune can't display the actual strikethrough yet.

Maybe even add a standard "strikethrough" face beside the bold, italic, and underlined ones.

Reply:
>You should still define the regex so it can be highlighted with a colour, even if Kakoune can't display the actual strikethrough yet.

Agree, I've commented it out to decide later.

>Maybe even add a standard "strikethrough" face beside the bold, italic, and underlined ones.

I want to extend all current themes with new standard faces for markup languages, like `heading`, `subheading`, `subsubheading`, `subsection`, `subsubsection`, `subsubsubsection`, and others, becouse in current state highlighting is limited too strictly.